src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -175,17 +158,42 @@ class MarathonSchedulerActor private (
       removeLocks(plan.affectedRunSpecIds)
       deploymentFailed(plan, reason)
 
-    case RunSpecScaled(id) => removeLock(id)
-
     case TasksKilled(runSpecId, _) => removeLock(runSpecId)
 
+    case StartInstance(runSpec) =>
+      if (noConflictsWith(Set(runSpec.id))) startInstance(runSpec)
+      else logger.info(s"Did not start an instance for ${runSpec.id} because runSpec is locked in a deployment.")
+
+    case DecommissionInstance(runSpec) =>
+      if (noConflictsWith(Set(runSpec.id))) decommissionInstance(runSpec)
+      else logger.info(s"Did not decommission an instance for ${runSpec.id} because runSpec is locked in a deployment.")
+
     case msg => logger.warn(s"Received unexpected message from ${sender()}: $msg")
   }
 
-  def scaleRunSpecs(): Unit = {
-    groupRepository.root().foreach { root =>
-      root.transitiveRunSpecs.foreach(spec => self ! ScaleRunSpec(spec.id))
+  private def startInstance(runSpec: RunSpec): Unit = {
+    launchQueue.add(runSpec, 1)
+    logger.debug(s"Successfully launched new instance for ${runSpec.id}")
+  }
+
+  private def decommissionInstance(runSpec: RunSpec): Future[Done] = async {
+    val runningInstances = await(instanceTracker.specInstances(runSpec.id)).filter(_.isActive)
+
+    def killToMeetConstraints(runningInstances: Seq[Instance], killCount: Int): Seq[Instance] =
+      Constraints.selectInstancesToKill(runSpec, runningInstances, killCount)
+
+    val ScalingProposition(instancesToKill, _) = ScalingProposition.propose(
+      runningInstances, None, killToMeetConstraints, runSpec.instances, runSpec.killSelection)
+
+    instancesToKill match {
+      case Some(i :: tail) =>
+        if (tail.nonEmpty) logger.warn(s"Expected to decommission exactly one instance for ${runSpec.id} but apparently there were more candidates available: ${tail.map(_.instanceId).mkString(",")}")
```

Comment:
hmm I don't get this log. Why is it warning? Why is it wrong that there are multiple instances?